### PR TITLE
[DSIP-68] Unify the table structure and table fields in backend and frontend ui sections

### LINKED ISSUE #16515: [DSIP-68] Uniformly name process in code as workflow
### Search before asking

- [X] I had searched in the [DSIP](https://github.com/apache/dolphinscheduler/issues/14102) and found no similar DSIP.

### Motivation

Uniformly name process in code as workflow for better maintenance

1. First modify the backend code uniformly #16516
3. Then unify the table structure and table fields in backend #16525
4. Unify the backend controller and frontend ui sections #16525

### Design Detail

Incompatible changes
- Rename tables and fields from `process` to `workflow`. eg, from `t_ds_process_definition` to `t_ds_workflow_definition`.
- Rename api request/response params from `process` to `workflow`. eg, from `processDefinitionCode` to `WorkflowDefinitionCode`.

These incompatible changes may affect users of secondary development based on API and Database.

### Compatibility, Deprecation, and Migration Plan

_No response_

### Test Plan

Test it with ut, it, e2e, api-test

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #16525: [Improvement] Unify the table structure and table fields in backend and frontend ui sections
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Unify the table structure and table fields in backend and frontend ui sections

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #16570: [Bug] [Api] WorkflowTaskLineage Param Passing Error
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

An error occurs when I attempt to create a workflow with a dependent node.
`Caused by: org.apache.ibatis.binding.BindingException: Parameter 'workflowTaskLineages' not found. Available parameters are [param1, processTaskLineages]`
Inconsistency in parameter names in the WorkflowTaskLineageMapper.

### What you expected to happen

None

### How to reproduce

None

### Anything else

None

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SbloodyS**
## Brief change log

close #16515
close #16525

- Add `@DisableIfTestFails` annotation to `e2e-test` and `api-test` to the test class, all tests after the first failed one are disabled for better debugging. [Junit](https://junit-pioneer.org/docs/disable-if-test-fails/)
- Schema-check CI improvement

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

## Pull Request Notice
[Pull Request Notice](https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: .github/workflows/backend.yml [hunk 1]
```diff
@@ -133,7 +133,9 @@ jobs:
           /bin/bash ${{ matrix.case.script }}
   schema-check:
     runs-on: ubuntu-latest
-    if: ${{ (needs.paths-filter.outputs.db-schema == 'true') || (github.event_name == 'push') }}
+    # Temporarily disable schema check until 3.3.0 is released
```
Comment (ruanwenjun) [inline .github/workflows/backend.yml]: It's needed to open the schema check, we should make sure the upgrade from 3.1.9, 3.2.0 can upgrade to 3.3.0.

File: dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/controller/ClusterControllerTest.java [hunk 2]
```diff
@@ -84,7 +84,7 @@ public void testCreateCluster() throws Exception {
         clusterCode = (String) result.getData();
     }
 
-    @Test
+    // @Test
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/controller/ClusterControllerTest.java]:
```suggestion
    @Test
```

File: dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/WorkflowE2ETest.java [hunk 3]
```diff
@@ -143,9 +145,9 @@ void testCreateSubWorkflow() {
                         .goToTab(WorkflowDefinitionTab.class);
 
         workflowDefinitionPage
-                .createSubProcessWorkflow()
+                .createSubWorkflowWorkflow()
```
Comment (ruanwenjun) [inline dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/WorkflowE2ETest.java]:
```suggestion
                .createSubWorkflowTask()
```

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/mysql/dolphinscheduler_ddl.sql [hunk 4]
```diff
@@ -126,3 +126,49 @@ CALL drop_column_t_ds_alert_plugin_instance;
...
+ALTER TABLE t_ds_process_task_relation change process_definition_version workflow_definition_version int(11);
+ALTER TABLE t_ds_process_task_relation change process_definition_code workflow_definition_code bigint(20);
+
+ALTER TABLE t_ds_process_task_relation_log change process_definition_version workflow_definition_version int(11);
+ALTER TABLE t_ds_process_task_relation_log change process_definition_code workflow_definition_code bigint(20);
+
+ALTER TABLE t_ds_process_instance change process_definition_code workflow_definition_code bigint(20);
+ALTER TABLE t_ds_process_instance change process_definition_version workflow_definition_version int(11);
+ALTER TABLE t_ds_process_instance change is_sub_process is_sub_workflow int(11);
+ALTER TABLE t_ds_process_instance change process_instance_priority workflow_instance_priority int(11);
```
Comment (ruanwenjun) [inline dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/mysql/dolphinscheduler_ddl.sql:158]: It's better to use  procedure to make sure the upgrade is idempotent

**Review by ruanwenjun — COMMENTED** [review]
We also need to upgrade the task params name of `SubWorkflowParameters` in DB.

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/mysql/dolphinscheduler_ddl.sql (hunk 4 above)
Comment (SbloodyS) [reply dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/mysql/dolphinscheduler_ddl.sql:158]: Done.

File: dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/WorkflowE2ETest.java (hunk 3 above)
Comment (SbloodyS) [reply dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/cases/WorkflowE2ETest.java]: Fixed.

File: dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/controller/ClusterControllerTest.java (hunk 2 above)
Comment (SbloodyS) [reply dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/controller/ClusterControllerTest.java]: Fixed.

File: .github/workflows/backend.yml (hunk 1 above)
Comment (SbloodyS) [reply .github/workflows/backend.yml]: Fixed.

File: .github/workflows/schema-check/postgresql/running-test.sh [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+export SPRING_DATASOURCE_PASSWORD="postgres"
+export SPRING_DATASOURCE_URL="jdbc:postgresql://127.0.0.1:5432/dolphinscheduler_dev"
+bash ds_schema_check_test/dev/tools/bin/upgrade-schema.sh
+
+# Install the target version schema and upgrade it
+export SPRING_DATASOURCE_URL="jdbc:postgresql://127.0.0.1:5432/dolphinscheduler_${DATABASE_VERSION}"
+bash ds_schema_check_test/${DS_VERSION}/tools/bin/upgrade-schema.sh
+bash ds_schema_check_test/dev/tools/bin/upgrade-schema.sh
+
+# Compare the schema
```
Comment (zhongjiajie) [inline .github/workflows/schema-check/postgresql/running-test.sh:36]: Thank you for migrating those into the file :+1:

File: docs/docs/zh/guide/upgrade/upgrade.md [hunk 6]
```diff
@@ -59,7 +59,7 @@ jar 包 并添加到 `./tools/libs` 目录下，设置以下环境变量
 
 执行结果：
 
-- 原血缘数据迁移至新血缘表 `t_ds_process_task_lineage`。
+- 原血缘数据迁移至新血缘表 `t_ds_workflow_task_lineage`。
```
Comment (zhongjiajie) [inline docs/docs/zh/guide/upgrade/upgrade.md:62]: Does upgrade.md versionable? or only have single version in our document just like faq.md?

**Review by zhongjiajie — DISMISSED** [review]
Great :+1: , only have one issue I am not sure

File: docs/docs/zh/guide/upgrade/upgrade.md (hunk 6 above)
Comment (SbloodyS) [reply docs/docs/zh/guide/upgrade/upgrade.md:62]: Yes. It's versionable.

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/mysql/dolphinscheduler_dml.sql [hunk 7]
```diff
@@ -14,3 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
 */
+
+UPDATE t_ds_task_definition SET task_type = 'SUB_WORKFLOW' WHERE task_type = 'SUB_PROCESS';
+UPDATE t_ds_task_definition_log SET task_type = 'SUB_WORKFLOW' WHERE task_type = 'SUB_PROCESS';
+UPDATE t_ds_task_definition SET task_params = replace(task_params, 'processDefinitionCode', 'workflowDefinitionCode') where task_type = 'SUB_WORKFLOW';
```
Comment (ruanwenjun) [inline dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/mysql/dolphinscheduler_dml.sql:20]: Also need to upgrade `t_ds_task_definition_log`.

 And Dynamic task need to upgrade too.

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/postgresql/dolphinscheduler_dml.sql [hunk 8]
```diff
@@ -14,3 +14,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
 */
+
+UPDATE t_ds_task_definition SET task_type = 'SUB_WORKFLOW' WHERE task_type = 'SUB_PROCESS';
+UPDATE t_ds_task_definition_log SET task_type = 'SUB_WORKFLOW' WHERE task_type = 'SUB_PROCESS';
+UPDATE t_ds_task_definition SET task_params = replace(task_params, 'processDefinitionCode', 'workflowDefinitionCode') where task_type = 'SUB_WORKFLOW';
```
Comment (ruanwenjun) [inline dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/postgresql/dolphinscheduler_dml.sql:20]: Same as above

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/mysql/dolphinscheduler_dml.sql (hunk 7 above)
Comment (SbloodyS) [reply dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/mysql/dolphinscheduler_dml.sql:20]: Fixed.

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/postgresql/dolphinscheduler_dml.sql (hunk 8 above)
Comment (SbloodyS) [reply dolphinscheduler-dao/src/main/resources/sql/upgrade/3.3.0_schema/postgresql/dolphinscheduler_dml.sql:20]: Fixed.

**Review by ruanwenjun — APPROVED** [review]

**Review by zhongjiajie — APPROVED** [review]

**Pull request merged**